workflows/src/test/kotlin/com/r3/corda/lib/tokens/workflows/CreateEvolvableTokenTests.kt

```diff
@@ -165,4 +166,65 @@ class CreateEvolvableTokenTests : JITMockNetworkTests() {
             alice.createEvolvableToken(token, notaryIdentity).getOrThrow()
         }
     }
+
+    // Flows with observers
+    @Test
+    fun `with unrelated observer`() {
+        // denise as observer
+        val token = factory.withOneMaintainerAndOneParticipant()
+
+        val createTx = alice.createEvolvableToken(token, notaryIdentity, listOf(denise.legalIdentity())).getOrThrow()
+        val createdToken = createTx.singleOutput<TestEvolvableTokenType>()
+
+        // Expect to have one create command with maintainer signature
+        val maintainerKeys = token.maintainers.map { it.owningKey }.toSet()
+        assertEquals(maintainerKeys, createTx.requiredSigningKeys, "Must be signed by all maintainers")
+
+        // Alice, Charlie, Denise should record the transaction
+        assertHasTransaction(createTx, alice, charlie, denise)
+        val aliceToken = alice.services.vaultService.queryBy<TestEvolvableTokenType>().states
+        assertEquals(createdToken, aliceToken.single())
+        val charlieToken = charlie.services.vaultService.queryBy<TestEvolvableTokenType>().states
+        assertEquals(createdToken, charlieToken.single())
+        val deniseQuery = denise.services.vaultService.queryBy<TestEvolvableTokenType>().states
+        assertEquals(createdToken, deniseQuery.single())
+    }
+
+    @Test
+    fun `with observer that is participant`() {
+        val token = factory.withOneMaintainerAndOneParticipant()
+
+        val createTx = alice.createEvolvableToken(token, notaryIdentity, listOf(charlie.legalIdentity())).getOrThrow()
+        val createdToken = createTx.singleOutput<TestEvolvableTokenType>()
+
+        // Expect to have one create command with maintainer signature
+        val maintainerKeys = token.maintainers.map { it.owningKey }.toSet()
+        assertEquals(maintainerKeys, createTx.requiredSigningKeys, "Must be signed by all maintainers")
+
+        // Alice, Charlie, Denise should record the transaction
+        assertHasTransaction(createTx, alice, charlie)
+        val aliceToken = alice.services.vaultService.queryBy<TestEvolvableTokenType>().states
+        assertEquals(createdToken, aliceToken.single())
+        val charlieToken = charlie.services.vaultService.queryBy<TestEvolvableTokenType>().states
+        assertEquals(createdToken, charlieToken.single())
+    }
+
+    @Test
+    fun `with observer that is maintainer`() {
```

Comment:
What's the difference between this test and the above one?

Reply:
It's confusing because I reused old states and tests to do that. Basically, observer is also maintainer of the state. In previous test observer is participant. They are different in terms of what I pass to the flow as an observers list.
I wanted to test that even if someone passes in observers that are already on the state (either as a maintainer or participant) the flows will still function correctly.
Maybe it needs some redesign though, I could use your input on how it should be used.

Reply:
ok thanks for the heads up - for now it's OK, I think.